### ISSUE TITLE
Post drafts should persist in memory when a user navigate between pages 

### ISSUE THREAD

**Issue by ivanminutillo** [bonfire-networks/bonfire-app#403]
We can simply cache the current draft or implement a "drafts", to be able to save multiple posts in memory?

**Comment by mayel** [comment]
We could also save it in on the client side: https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage

**Comment by abaez** [comment]
I think the session storage on client side, as @mayel suggested, would be a good way to go. Would help in keep strain on memory and complexity to a minimum. Especially as larger instances hopefully spin up for bonfire.

If no one minds, I would like to stab at this. It's bothering me each time I navigate with a working draft. 😅

**Comment by mayel** [comment]
@abaez You're more than welcome to! ❤️ This is probably the place to start: https://github.com/bonfire-networks/bonfire_editor_quill/blob/main/assets/js/extension.js

**Comment by mayel** [comment]
You'll also want to have change the path to that file here to point to your local clone during development : https://github.com/bonfire-networks/bonfire-app/blob/main/flavours/classic/config/deps_hooks.js

**Comment by abaez** [comment]
@mayel thanks for the help. I already set myself locally for tweaking. 

Though got sidetracked in trying to get bonfire to build with nix. Looks like it wasn't maintained for a while. Planning to spend the weekend to see if I can make it related to the whole Just workflow you all have going. So it ends up updating anytime the related environment variables are updated.

**Comment by mayel** [comment]
@abaez yeah the nix setup was contributed a while back but never used or tested much... did you also see option D in the hacking docs? no docker is required

**Comment by abaez** [comment]
@mayel I did and is why I went the Nix approach. It's bare metal without polluting my system. Though it's not like I _don't_ already have rust, yarn, or elixir installed on my local... 😛

I do like being able to have idempotent build environments. For all those too many times of things going haywire and not have to spend hours getting the dev environment just how I need it. Containerizing does this to a degree (only if you pay attention to package lock your whole build). Nix provides a way to do that with only identifying it once. The set up here seems it was to focus mostly on nixos rather than to make it as synergistic with the current build workflow.... I'm going to make a new issue to capture this side work. 😅

**Comment by abaez** [comment]
For now, I'll see how far I get with #405 during the weekend. I'll  also probably write the fix to this in the meantime. I already got an idea what is needed here and can begin building it out before I have a test environment to test. 😂

**Comment by mayel** [comment]
@abaez is there anything we can do to help progress this issue or #405 ?

**Comment by abaez** [comment]
@mayel sorry about the silence. Been pretty preoccupied with my day job lately.😔 I had something already semi functioning. But again, haven't touched since pushing up the draft.

Will try to wrap both of these this weekend.

**Comment by mayel** [comment]
No worries, thanks again for looking at it! BTW there's probably been some changes to the quill JS integration you'll need to merge.

**Comment by mayel** [comment]
This has now been partly implemented by using a LV sticky view (draft contents aren't actually saved atm, but at least don't disappear when you navigate from page to page within bonfire)

**Comment by abaez** [comment]
@mayel completely forgot about covering this! Terribly sorry.

I can do the remaining part of getting any nuances done with the new updates. Let me know and I'll get on it this coming week. Again, sorry for missing on covering here.

**Comment by mayel** [comment]
No worries @abaez, I think we can put saving drafts in the backlog, and change the scope to a feature that saves (multiple) drafts in the backend (so it's available to the user across browsers/devices) since we achieved most of the goal of this issue with that LiveView trick. A couple things I'd still like to figure out within this scope are:

- [ ] show the "Your draft will be lost. Continue?" when you close the modal only if something was actually drafted
- [ ] maybe show a similar popup when navigating to a non-LiveView page (in the same tab) which would still result in the draft being lost, something like `onbeforeunload` but on the form rather than the window

**Comment by abaez** [comment]
> No worries @abaez <https://github.com/abaez>, I think we can put saving drafts in the backlog, and change the scope to a feature that saves (multiple) drafts in the backend (so it's available to the user across browsers/devices) since we achieved most of the goal of this issue with that LiveView trick.
> 
For some clarity, do you mean to change this issue scope to saving drafts to the backend? I can do the change, but want to make sure I understood if you want this for #403 or if you meant a separate issue to spawn. 

> 
> 
> A couple things I'd still like to figure out within this scope are:
>  • show the "Your draft will be lost. Continue?" when you close the modal only if something was actually drafted
>  • maybe show a similar popup when navigating to a non-LiveView page (in the same tab) which would still result in the draft being lost, something like `onbeforeunload` but on the form rather than the window
 I can take a swing in discovery for these at the very least.

**Comment by mayel** [comment]
Yeah I meant a new backlog issue for savings of  drafts in DB 👍

**Comment by mayel** [comment]
we're now doing this with a persistent child LiveView, and may extend that further with #1489
